.github/workflows/_docs.yml

```diff
@@ -39,6 +39,7 @@ jobs:
     # Don't run on forked repos.
     if: github.repository_owner == 'pytorch'
     runs-on: ${{ matrix.runner }}
+    environment: ${{ github.ref == 'refs/heads/main' && 'update-commit-hash' || '' }}
```

Comment:
Hi Catherine, not sure this will work  since GH_PYTORCHBOT_TOKEN sometimes passed from top level workflow:
```
 docs-push:
    name: docs push
    uses: ./.github/workflows/_docs.yml
    needs: docs-build
    with:
      build-environment: linux-jammy-py3.8-gcc11
      docker-image: ${{ needs.docs-build.outputs.docker-image }}
      push: ${{ github.event_name == 'schedule' || startsWith(github.event.ref, 'refs/tags/v') }}
      run-doxygen: true
    secrets:
      GH_PYTORCHBOT_TOKEN: ${{ secrets.GH_PYTORCHBOT_TOKEN }}
```

Please see this Nikita's comment: https://github.com/pytorch/pytorch/pull/107060#pullrequestreview-1574507112

Reply:
I tested on a dummy repo and the job passed